opm/grid/cpgrid/CpGrid.cpp

```diff
@@ -245,21 +245,94 @@ CpGrid::scatterGrid(EdgeWeightMethod method, const std::vector<cpgrid::OpmWellTy
-        int num_cells = distributed_data_->cell_to_face_.size();
 
-        std::vector<int> cc_num_cells(cc.size(), 0);
-        cc_num_cells[cc.rank()] = num_cells;
-        cc.sum(cc_num_cells.data(),cc.size());
+
+        // Compute the partition type for cell
+        int owned_cells = 0;
+        int overlap_cells = 0;
+        for (const auto i : distributed_data_->cell_indexset_) {
+            if (i.local().attribute() == AttributeSet::owner) {
+                ++owned_cells;
+            } else {
+                ++overlap_cells;
+            }
+        }
+
+        // owned cells
+        std::vector<int> cc_owned_cells(cc.size(), 0);
+        cc_owned_cells[cc.rank()] = owned_cells;
+        cc.sum(cc_owned_cells.data(), cc.size());
+
+        // overlap cells
+        std::vector<int> cc_overlap_cells(cc.size(), 0);
+        cc_overlap_cells[cc.rank()] = overlap_cells;
+        cc.sum(cc_overlap_cells.data(), cc.size());
+
+        // total cells
+        const int total_cells = distributed_data_->cell_to_face_.size();
+        std::vector<int> cc_total_cells(cc.size(), 0);
+        cc_total_cells[cc.rank()] = total_cells;
+        cc.sum(cc_total_cells.data(), cc.size());
 
         if (cc.rank() == 0) {
-            std::string str = "\nAfter loadbalancing on " + std::to_string(cc.size()) + " processes we have\nrank: num cells\n----------------\n";
+            std::string str = "\nLoad balancing distributes " + std::to_string(data_->size(0))
+                            + " active cells on " + std::to_string(cc.size()) + " processes as follows:\n";
+            str += "  rank   owned cells   overlap cells   total cells\n";
+            str += "--------------------------------------------------\n";
             for (int i = 0; i < cc.size(); ++i) {
-                str += std::to_string(i) + ": " + std::to_string(cc_num_cells[i]) + "\n";
+                // rank
+                {
+                    auto str_tmp = std::to_string(i);
+                    str_tmp.insert(str_tmp.begin(), 6 - str_tmp.length(), ' ');
+                    str += str_tmp;
+                }
+                // owned cells
+                {
+                    auto str_tmp = std::to_string(cc_owned_cells[i]);
+                    str_tmp.insert(str_tmp.begin(), 14 - str_tmp.length(), ' ');
+                    str += str_tmp;
+                }
+                // overlap cells
+                {
+                    auto str_tmp = std::to_string(cc_overlap_cells[i]);
+                    str_tmp.insert(str_tmp.begin(), 16 - str_tmp.length(), ' ');
+                    str += str_tmp;
+                }
+                // total cells
+                {
+                    auto str_tmp = std::to_string(cc_total_cells[i]);
+                    str_tmp.insert(str_tmp.begin(), 14 - str_tmp.length(), ' ');
+                    str += str_tmp;
+                }
+                str += "\n";
```

Comment:
I'm sure this is correct, and the output certainly looks good, but I worry about the maintainability of these constructs.  Did you consider using `ostringstream` and `iomanip` here?  For instance, something along the lines of
```C++
std::ostringstream line;

line << std::setw(6) << i
     << std::setw(14) << cc_owned_cells[i]
     << std::setw(16) << cc_overlap_cells[i]
     << std::setw(14) << cc_total_cells[i] << '\n';

str += line.str();
```
possibly with some additional constants to define the field widths?

Reply:
Looks better. I will change it.